### PR TITLE
Rename HttpImage to HttpImageProvider

### PR DESCRIPTION
**Pull request by brianquinlan**

### CODE REVIEW

File: http_image_provider/lib/http_image_provider.dart [hunk 1]
```diff
@@ -119,3 +124,6 @@ class HttpImage extends ImageProvider<HttpImage> {
   String toString() =>
       '${objectRuntimeType(this, 'HttpImage')}("$url", scale: $scale)';
```
Comment (ueman) [inline http_image_provider/lib/http_image_provider.dart]: I guess this should be renamed too
```suggestion
      '${objectRuntimeType(this, 'HttpImageProvider')}("$url", scale: $scale)';
```

File: http_image_provider/lib/http_image_provider.dart [hunk 2]
```diff
@@ -119,3 +124,6 @@ class HttpImage extends ImageProvider<HttpImage> {
...
 }
+
+@Deprecated('Use HttpClientImageProvider instead')
```
Comment (ueman) [inline http_image_provider/lib/http_image_provider.dart]:
```suggestion
@Deprecated('Use HttpImageProvider instead')
```

File: http_image_provider/pubspec.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ environment:
   flutter: ">=3.0.0"
 
 dependencies:
-  http: ^1.1.0
+  http: ^1.2.0
```
Comment (ueman) [inline http_image_provider/pubspec.yaml:17]: I don't mind this change, but is there a reason to force consumers to use a more recent version?

File: http_image_provider/test/http_image_provider_test.dart
Comment (ueman) [inline http_image_provider/test/http_image_provider_test.dart:1]: This is awesome, thank you!

**Review by ueman — COMMENTED** [review]
Thank you! I just have two small nitpicks, but it's otherwise good to go 🚀

File: http_image_provider/test/http_image_provider_test.dart
Comment (brianquinlan) [reply http_image_provider/test/http_image_provider_test.dart:1]: It's not a great test and I didn't add workflow to it but we can add more to it over time.

And I should be thanking you for this package!

File: http_image_provider/pubspec.yaml (hunk 3 above)
Comment (brianquinlan) [reply http_image_provider/pubspec.yaml:17]: 1.2.0 has a test helper that I'm using. I could just add that constraint to the `dev_dependencies`, if you'd prefer. 1.2.0 has been out for about 5 months.

**Comment by ueman** [comment]
It's now published: https://pub.dev/packages/http_image_provider/versions

**Pull request merged**